spec/index.bs

```diff
@@ -163,6 +197,18 @@ spec: WebIDL; urlPrefix: https://heycam.github.io/webidl/#
   A <a>sync registration</a> has an associated <a>service worker registration</a>. It is initially set to null.
 
   Within one <a>list of sync registrations</a> each <a>sync registration</a> MUST have a unique <a>tag</a>.
+
+  A <dfn>periodic sync registration</dfn> is a tuple consisting of a <a>tag</a>, a dictionary of <a>options</a>, and a <a lt="registration state">state</a>.
+
+  A <a>periodic sync registration</a> has an associated <a>tag</a>, a DOMString.
+
+  A <a>periodic sync registration</a> has an associated <dfn>options</dfn>, a dictionary containing <dfn>minInterval</dfn>, a long long. <a>minInterval</a> is used to specify the minimum interval, in milliseconds, at which the periodic synchronization should happen.
+
+  A <a>periodic sync registration</a> has an associated <a>registration state</a>, which is one of <a>pending</a>, <a>waiting</a>, <a>firing</a>, or <a>reregisteredWhileFiring</a>. It is initially set to <a>pending</a>.
```

Comment:
Seems like you can define the enum once and reference it.

Does reregisteredWhileFiring make sense in the context of periodic syncs though?

Reply:
That's exactly what I'm doing. Note the <dfn>s on line 194, and the later ones are simply references -- <a>

I think it does -- we'll update the existing periodicSync registration with new options if another register() is invoked later for a pre-existing tag. At that point, the existing one might be firing.

Reply:
Yes I noted that, I don't like that it is referencing a definition in the definition, then reiterating the possible values.

How about moving the enum definition to a single line, then just say that a registration state is one of those for both one-shot and periodic.

> I think it does -- we'll update the existing periodicSync registration with new options if another register() is invoked later for a pre-existing tag. At that point, the existing one might be firing.

I don't think that makes sense. With the one-shot version, we need to reregister it for the case you descibed otherwise it won't be clear to developers whether a registration went through or not. Since we don't delete periodic registrations after their events have been fired, this will never be the case.

Reply:
But options will be updated, meaning that the expected frequency of periodicSync events will change. It's good to notify the developer that this change happened when one event was being fired. WDYT?

Reply:
I personally disagree, because even with the change future syncs won't be lost. So I guess just go ahead with this and we can discuss further on a spec issue.